api/src/routers/meals.js

```diff
@@ -20,7 +20,128 @@ function handleFormatDateOrDatetime(fieldToFormat, value, format, res) {
 //  ----------- /api/meals | GET | Returns all meals -----------
 meals.get("/", async (req, res) => {
   try {
-    const meals = await knex("Meal").select("*");
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir,
+    } = req.query;
+    let meals = await knex("Meal").select("*");
+    let query = knex("Meal")
+      .leftJoin(
+        knex.raw(`(
+        SELECT meal_id, COALESCE(SUM(number_of_guests), 0) AS total_guests
+        FROM Reservation
+        GROUP BY meal_id
+      ) AS Reserved`),
+        "Meal.id",
+        "Reserved.meal_id"
+      )
+      .select(
+        "Meal.id",
+        "Meal.title",
+        "Meal.max_reservations",
+        knex.raw("COALESCE(Reserved.total_guests, 0) AS total_guests"),
+        "Meal.price",
+        "Meal.when"
+      );
+
+    // maxPrice Parameter
+    if (maxPrice) {
+      query = query.where("Meal.price", "<=", maxPrice);
+    }
+
+    // availableReservations Parameter
+    if (availableReservations !== undefined) {
```

Comment:
for this kind of complex checks in the api that may take many levels I would encourage you to checkout out [early returns patter](https://medium.com/swlh/return-early-pattern-3d18a41bba8). It will make your code more legible and prevent excessive nesting!

Reply:
Thank you so much for your suggestion. I've update the files with 'return early patter' :)